server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -810,7 +810,7 @@ public class ManagementServerImpl extends ManagerBase implements ManagementServe
     protected StateMachine2<State, VirtualMachine.Event, VirtualMachine> _stateMachine;
 
     static final ConfigKey<Integer> vmPasswordLength = new ConfigKey<Integer>("Advanced", Integer.class, "vm.password.length", "6", "Specifies the length of a randomly generated password", false);
-    static final ConfigKey<String> vmPasswordComplexity = new ConfigKey<String>("Advanced", String.class, "vm.password.complexity", "^(?=.{6,})(?=.*[A-Z])(?=.*[a-z])(?=.*[@#%&^~.,!,?,:,;])", "Specifies the pattern for generated password", false);    
+    static final ConfigKey<String> vmPasswordComplexity = new ConfigKey<String>("Advanced", String.class, "vm.password.complexity", ".*", "Specifies the pattern for generated password", true);    
```

Comment:
.* is possibly more complex and possibly simpler than what is generated, this is ok though. the generated one is 6 chars with at least lower case, upper case and numeral.

Reply:
sorry if i am wrong but if you are referring to passwordgenerator.java file i have added code to add special characters too

and the reasoni used .* is 
in above conversation this was told so
@Pavan-Nambi, to keep compatibility with the current behavior (accept any password generated), the default value should be .*; then, one should change the configuration via API/UI according to the necessity.

Reply:
@DaanHoogland, the point of suggesting value `.*`  was to keep the current behavior easily; however, it can be achieved with other approaches, like handling an empty pattern.

Reply:
Ah, right, thanks @GutoVeronezi . This looks to me that any password would be allowed. If we leave it at this this behaviour should be well documented.